### ISSUE TITLE
[Event Request] Table 1003 "Job Planning Line" - OnBeforeUpdateRemainingCostsAndAmounts

### ISSUE THREAD

**Issue by LewisHyettTES** [microsoft/ALAppExtensions#24808]
Could you please add the following OnBeforeUpdateRemainingCostsAndAmounts event to procedure UpdateRemainingCostsAndAmountsin Table 1003 "Job Planning Line"?

Current
```
local procedure UpdateRemainingCostsAndAmounts(PostingDate: Date; CurrencyFactor: Decimal)
    begin
        if "Usage Link" then begin
            InitRoundingPrecisions();
            "Remaining Total Cost" := Round("Unit Cost" * "Remaining Qty.", AmountRoundingPrecisionFCY);
            "Remaining Total Cost (LCY)" :=
                ConvertAmountToLCY(
                    PostingDate, "Remaining Total Cost", CurrencyFactor, AmountRoundingPrecision);
            "Remaining Line Amount" := CalcLineAmount("Remaining Qty.");
            "Remaining Line Amount (LCY)" :=
                ConvertAmountToLCY(
                    PostingDate, "Remaining Line Amount", CurrencyFactor, AmountRoundingPrecision);
        end else
            ClearValues();
    end;
```

Requested

```
local procedure UpdateRemainingCostsAndAmounts(PostingDate: Date; CurrencyFactor: Decimal)
var
	IsHandled: Boolean;
begin
       IsHandled := false;
	OnBeforeUpdateRemainingCostsAndAmounts(Rec, PostingDate, CurrencyFactor,AmountRoundingPrecisionFCY, AmountRoundingPrecision, IsHandled);
	if IsHandled then
		exit;
	if "Usage Link" then begin
		InitRoundingPrecisions();
		"Remaining Total Cost" := Round("Unit Cost" * "Remaining Qty.", AmountRoundingPrecisionFCY);
		"Remaining Total Cost (LCY)" :=
			ConvertAmountToLCY(
				PostingDate, "Remaining Total Cost", CurrencyFactor, AmountRoundingPrecision);
		"Remaining Line Amount" := CalcLineAmount("Remaining Qty.");
		"Remaining Line Amount (LCY)" :=
			ConvertAmountToLCY(
				PostingDate, "Remaining Line Amount", CurrencyFactor, AmountRoundingPrecision);
	end else
		ClearValues();
end;
```

**Additional context**

We would like to adjust the way these fields are calculated which currently is not possible.

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 12576.

**Comment by LewisHyettTES** [comment]
@JesperSchulz Is there any update to when this event will be released?

**Comment by JesperSchulz** [comment]
It should have been released with 23.1 already.
